<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>yoloControl</class>
 <widget class="QMainWindow" name="yoloControl">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1081</width>
    <height>353</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>简易训练工具</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_5">
    <item>
     <widget class="QSplitter" name="splitter_2">
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
      <widget class="QWidget" name="">
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="0" colspan="2">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>请选择您的YOLO版本: </string>
            </property>
            <property name="alignment">
             <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QComboBox" name="m_comVersion">
            <item>
             <property name="text">
              <string>YOLOv5</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>YOLOv8</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>YOLOv11</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>使用的conda环境名称是:</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="1" column="2" colspan="2">
           <widget class="QComboBox" name="m_comEvnName">
            <property name="editable">
             <bool>true</bool>
            </property>
            <item>
             <property name="text">
              <string>yolo_to_kmodel</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>打开YOLO根目录:</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="2" column="1" colspan="2">
           <widget class="QLineEdit" name="m_lineYoloPath"/>
          </item>
          <item row="2" column="3">
           <widget class="QPushButton" name="m_btnYoloOpen">
            <property name="text">
             <string>打开</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>打开配置文件:</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="3" column="1" colspan="2">
           <widget class="QLineEdit" name="m_lineYamlPath"/>
          </item>
          <item row="3" column="3">
           <widget class="QPushButton" name="m_btnYamlOpen">
            <property name="text">
             <string>打开</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QLabel" name="label_4">
            <property name="frameShadow">
             <enum>QFrame::Shadow::Sunken</enum>
            </property>
            <property name="text">
             <string>配置训练参数:</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignmentFlag::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QGridLayout" name="gridLayout_3">
            <item row="0" column="0">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Epochs:</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="m_spinEpochs">
              <property name="maximum">
               <number>1000000</number>
              </property>
              <property name="value">
               <number>250</number>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Batch Size:</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="m_spinBatchSize">
              <property name="maximum">
               <number>1000000</number>
              </property>
              <property name="value">
               <number>8</number>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>Image Size:</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QSpinBox" name="m_spinImageSize">
              <property name="maximum">
               <number>1000000</number>
              </property>
              <property name="value">
               <number>320</number>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_8">
              <property name="text">
               <string>Devic:</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QComboBox" name="m_comboDevice">
              <item>
               <property name="text">
                <string>GPU</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>CPU</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="1">
           <widget class="QPushButton" name="m_btnTrainPt">
            <property name="text">
             <string>开始训练</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QPushButton" name="m_btnStop">
            <property name="text">
             <string>停止训练</string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="1" column="0">
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="m_btn2Onnx">
            <property name="text">
             <string>toOnnx</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="m_btn2Kmodel">
            <property name="text">
             <string>toKmodel</string>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <spacer name="horizontalSpacer_4">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QSplitter" name="splitter">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <widget class="QWidget" name="">
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Orientation::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="label_11">
           <property name="text">
            <string>不训练直接转化需要提供pt文件和val数据集</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="0" column="0" colspan="2">
            <widget class="QLabel" name="label_12">
             <property name="text">
              <string>pt文件路径:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2" colspan="3">
            <widget class="QLineEdit" name="m_linePtPath"/>
           </item>
           <item row="0" column="5">
            <widget class="QPushButton" name="m_btnPtOpen">
             <property name="text">
              <string>打开</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="2">
            <widget class="QLabel" name="label_13">
             <property name="text">
              <string>val数据集路径:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2" colspan="3">
            <widget class="QLineEdit" name="m_lineValPath"/>
           </item>
           <item row="1" column="5">
            <widget class="QPushButton" name="m_btnValOpen">
             <property name="text">
              <string>打开</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <spacer name="horizontalSpacer_5">
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="2" column="1" colspan="2">
            <widget class="QPushButton" name="m_btnDir2Onnx">
             <property name="text">
              <string>toOnnx</string>
             </property>
            </widget>
           </item>
           <item row="2" column="3">
            <widget class="QPushButton" name="m_Dir2Kmodel">
             <property name="text">
              <string>toKmodel</string>
             </property>
            </widget>
           </item>
           <item row="2" column="4" colspan="2">
            <spacer name="horizontalSpacer_6">
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Orientation::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="">
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>打印日志信息</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QTextEdit" name="m_editLog">
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
